docs/ops/doc/CallingOps.md

```diff
@@ -1,61 +1,96 @@
+Calling our Gaussian blur as a *computer* above is great when we have pre-allocated output, but for other scenarios we can request Ops as *functions* or *inplaces*.
+
+*Functions* are used when we want to *create* the final output, indicated by ending the builder with `.apply()`:
+
+```groovy
+var outImage = ops.op("filter.gauss").input(inImage, 2.0).apply()
+```
+
+*Inplaces* are used when we want to destructively modify one of the existing inputs (which is explicitly forbidden by *computers*; a *computer* Op's output should be a different object from all of its inputs). We indicate this by the `mutate#()` method, where the `#` corresponds to the *parameter index* that will be modified:
+
+```
+# Modify the first input in-place
+ops.op("filter.gauss").input(inImage, 2.0).mutate1()
+```
+
+Note that although the final method call changes for each mode of operation, *this is based on the path taken through the `OpBuilder` chain*. For example, we cannot call the `compute()` method if we haven't provided an `.output()`:
+
+```
+# Does not compute
+ops.op("filter.gauss").input(inImage, 2.0).compute()
+```
+
+A key takeaway from this section is that how you **request** the Op does not necessarily need to match how the Op is **implemented**. *Functions* and *computers* should be largely interchangeable, thanks to the Ops engine's adaptation subsystem. For the 1.0.0 release we do not have the necessary adapters to go between *inplaces* and the other paradigms, but it is on our [development roadmap](https://github.com/scijava/scijava/issues/47)!
+
+### Repeating execution
+
+When you want to call an Op many times on different inputs, the `OpBuilder` can be used to return the *Op* itself, instead of performing the computation. Instead of calling the `.compute()` function at the end of our `OpBuilder` chain, we can use the `.computer()` method (or `.inplace()` or `.function()`, as appropriate) to get back the matched Op, which can then be reused via its `.compute()` method (or `.apply()` or `.mutate#()`, respectively):
 
 ```groovy
 var gaussOp = ops.op("filter.gauss").input(inImage, 2.0).output(outImage).computer()
-gaussOp.compute(inImage, 2.0, outImage)
+gaussOp.compute(inImage, 2.0, outImage1)
+gaussOp.compute(inImage, 5.0, outImage2)
 ```
 
+While we do pass concrete inputs and outputs in this example, they are essentially just being used to reason about the desired *types* - which we'll cover in the next section.
+
 *Note that the default `OpEnvironment` implementations cache Op requests* - this means that repeated `OpBuilder` requests targeting the same action will be faster than the original matching call.
 
-## Additions: Matching with classes
+### Matching with classes
```

Comment:
Is `classes` better or worse than `types`?

Reply:
3648c07bd304517d3dd16f8fc547b0ebd28e8b72